src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5833,9 +5835,10 @@ private bool IsUsingAliasInScope(string name)
             }
         }
 
-        private static void WarnOnAccessOfOffDefault(SyntaxNode node, BoundExpression boundLeft, DiagnosticBag diagnostics)
+        private void WarnOnAccessOfOffDefault(SyntaxNode node, BoundExpression boundLeft, DiagnosticBag diagnostics)
         {
-            if (boundLeft != null && boundLeft.Kind == BoundKind.DefaultExpression && boundLeft.ConstantValue == ConstantValue.Null)
+            if (boundLeft != null && boundLeft.Kind == BoundKind.DefaultExpression && boundLeft.ConstantValue == ConstantValue.Null &&
+                Compilation.Options.NullableContextOptions == NullableContextOptions.Disable)
```

Comment:
>Compilation.Options.NullableContextOptions == NullableContextOptions.Disable [](start = 16, length = 76)

I am not sure why are we checking NullableContextOptions on a compilation here. The nullable warnings or both nullable contexts can still be enabled locally in the file. Please make sure we have tests to cover both scenarios. If we want to check language version, as this [https://github.com/dotnet/roslyn/issues/34855#issuecomment-481055524](https://github.com/dotnet/roslyn/issues/34855#issuecomment-481055524) suggests, then we should check the language version used by the syntax tree instead.